File: source/oddProduct/baseline.js
const REPEATS = 100;
const n = Number(process.argv[2]);

const now = Date.now();

let repeat = REPEATS;

while (repeat--) {
  let product = 1;
  for (let i = 1; i <= n; i++) {
    product = (product * (2 * i - 1)) & 0xffffffff;
  }

  if (repeat == 0) {
    console.log(`Result: ${product < 0 ? product + 2 ** 32 : product}`);
  }
}

console.log(`Used time (average, in ms): ${(Date.now() - now) / REPEATS}`);

File: source/oddProduct/baseline.rs
use std::time::Instant;

pub fn odd_product(n: u32) -> u64 {
    (1..=n as u64).fold(1, |prod, x| prod.wrapping_mul(2 * x - 1))
}

const REPEAT: u128 = 1000;

fn main() {
    let args: Vec<String> = std::env::args().collect();

    let n: u32 = args[1].parse().unwrap();
    let now = Instant::now();

    let mut a = 0;

    for _ in 0..REPEAT {
        a = odd_product(n)
    }

    println!(
        "Result: {},\nUsed time (average, in milliseconds): {}",
        a,
        (now.elapsed().as_nanos() / REPEAT) as f32 / 1_000_000.0
    );
}
